src/rgw/rgw_orphan.cc

```diff
@@ -277,9 +277,28 @@ int RGWOrphanSearch::build_all_oids_index()
     string oid = i->get_oid();
     string locator = i->get_locator();
 
-    string name = oid;
-    if (locator.size())
-      name += " (@" + locator + ")";  
+    ssize_t pos = oid.find('_');
+    if (pos < 0) {
+      cout << "unidentified oid: " << oid << ", skipping" << std::endl;
+      /* what is this object, oids should be in the format of <bucket marker>_<obj>,
+       * skip this entry
+       */
+      continue;
+    }
+    string stripped_oid = oid.substr(pos + 1);
```

Comment:
I'm concerned about the +1 here to skip over the '\_'. The first check in parse_raw_oid() is
```c++
    if (oid[0] != '_') {
      *obj_name = oid;
      return true;
    }
```
So it looks to me like stripping "\<bucket\>\_" from the beginning will prevent parse_raw_oid() from ever parsing a namespace, whereas leaving that trailing '\_' should work as intended.